### PR TITLE
Auto-generation of PoX contract docs

### PR DESCRIPTION
**Pull request by kantai**
This PR adds `vm::docs::contracts`, which supports (somewhat) auto-generating JSON documentation for boot code contracts. Currently, this is just documentation for the `pox.clar` contract.

It does this by:

* generating a signature, input, and output types from the type checker for public and read-only functions
* evaluating any error code constants to generate error code documentation
* requiring a "description" for each published public and read-only function defined

This also adds the new JSON documentation to the PR robot (as well as changing the PR robot's git email address so that it can sign the CLA).

### CODE REVIEW

**Comment by kantai** [comment]
Built docs: https://gist.github.com/kantai/17fe06eacdf1864c7df490bb17c771da

**Review by lgalabru — APPROVED** [review]
Looks good to me!

File: src/vm/docs/contracts.rs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    value_type: String,
+    value: String,
+}
+
+struct ContractSupportDocs {
+    descriptions: HashMap<&'static str, &'static str>,
+    skip_func_display: HashSet<&'static str>,
+}
+
+fn make_contract_support_docs() -> HashMap<&'static str, ContractSupportDocs> {
```
Comment (jcnelson) [inline src/vm/docs/contracts.rs:40]: This is fine for now, but do we want to extract the comments directly from the PoX contract?  Maybe there could be a docstring annotation -- e.g. `;;;` indicates a docstring in Clarity?

**Review by jcnelson — APPROVED** [review]
LGTM!

File: src/vm/docs/contracts.rs (hunk 1 above)
Comment (kantai) [reply src/vm/docs/contracts.rs:40]: Yeah, that would be a nice feature, but would require some reworking of the parser (it currently doesn't parse comments at all). Will open an issue for that.

**Pull request merged**